chainerx_cc/chainerx/cuda/cuda_device/misc.cu

```diff
@@ -62,6 +62,50 @@ public:
 
 CHAINERX_REGISTER_OP_CUDA(SqrtOp, CudaSqrtOp);
 
+template <typename T>
+struct FabsImpl {
+    using CudaType = cuda_internal::DataType<T>;
+    __device__ void operator()(int64_t /*i*/, CudaType x, CudaType& out) { out = cuda::Fabs(x); }
+};
+
+class CudaFabsOp : public FabsOp {
+public:
+    void Call(const Array& x, const Array& out) override {
+        Device& device = x.device();
+        device.CheckDevicesCompatible(x, out);
+        CudaSetDeviceScope scope{device.index()};
+        const Array& x_cast = x.dtype() == out.dtype() ? x : x.AsType(out.dtype());
+        VisitFloatingPointDtype(out.dtype(), [&](auto pt) {
+            using T = typename decltype(pt)::type;
+            Elementwise<const T, T>(FabsImpl<T>{}, x_cast, out);
+        });
+    }
+};
+
+CHAINERX_REGISTER_OP_CUDA(FabsOp, CudaFabsOp);
+
+template <typename T>
+struct SignImpl {
+    using CudaType = cuda_internal::DataType<T>;
+    __device__ void operator()(int64_t /*i*/, CudaType x, CudaType& out) { out = cuda::Sign(x); }
+};
+
+class CudaSignOp : public SignOp {
+public:
+    void Call(const Array& x, const Array& out) override {
+        Device& device = x.device();
+        device.CheckDevicesCompatible(x, out);
+        CudaSetDeviceScope scope{device.index()};
+        const Array& x_cast = x.dtype() == out.dtype() ? x : x.AsType(out.dtype());
+        VisitNumericDtype(out.dtype(), [&](auto pt) {
+            using T = typename decltype(pt)::type;
+            Elementwise<const T, T>(SignImpl<T>{}, x_cast, out);
+        });
+    }
+};
+
+CHAINERX_REGISTER_OP_CUDA(SignOp, CudaSignOp);
```

Comment:
Ditto. We renamed it to `CHAINERX_CUDA_REGISTER_OP`.

Reply:
I guess I'll wait till the #6944 goes in.